client/blocks/app-promo/index.jsx

```diff
@@ -1,55 +1,85 @@
 import { Card, Button } from '@automattic/components';
 import { useLocalizeUrl } from '@automattic/i18n-utils';
+import classNames from 'classnames';
 import { useTranslate, useRtl } from 'i18n-calypso';
 import wpToJpImageRtl from 'calypso/assets/images/jetpack/wp-to-jp-rtl.svg';
 import wpToJpImage from 'calypso/assets/images/jetpack/wp-to-jp.svg';
+import QrCode from 'calypso/blocks/app-promo/qr-code';
 import AppsBadge from 'calypso/blocks/get-apps/apps-badge';
 import CardHeading from 'calypso/components/card-heading';
 import userAgent from 'calypso/lib/user-agent';
-
+import { useSelector } from 'calypso/state';
+import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
 import './style.scss';
 
-export const AppPromo = () => {
+export const AppPromo = ( {
+	title = false,
+	iconSize = 29,
+	source = 'calypso-customer-home',
+	subheader = false,
+	className = 'customer-home__card',
+	hasQRCode = false,
+} ) => {
 	const isRtl = useRtl();
 	const translate = useTranslate();
 	const localizeUrl = useLocalizeUrl();
-	const { isDesktop, isiPad, isiPod, isiPhone, isAndroid } = userAgent;
+	const isLoggedIn = useSelector( isUserLoggedIn );
+	const iconWidth = Math.ceil( ( 49 / 29 ) * iconSize );
+
+	const { isDesktop, isiPad, isiPod, isiPhone, isAndroid, isMobile } = userAgent;
 	const isIos = isiPad || isiPod || isiPhone;
 
-	const showIosBadge = ! isAndroid && ! isDesktop;
-	const showAndroidBadge = ! isIos && ! isDesktop;
-	const showBadge = showIosBadge || showAndroidBadge;
+	const showIosBadge = ! isAndroid && ! isDesktop && userAgent;
+	const showAndroidBadge = ! isIos && ! isDesktop && userAgent;
+	const showBadge = ( showIosBadge || showAndroidBadge ) && isMobile;
+	if ( title === false ) {
+		title = translate( 'Get our mobile app' );
+	}
+
+	if ( subheader === false ) {
+		subheader = translate( 'Everything you need to build and grow your site from any device.' );
+	}
+
+	className = classNames( className, 'app-promo' );
 
 	return (
-		<Card className="app-promo customer-home__card">
+		<Card className={ className }>
 			<img
 				className="app-promo__icon"
 				src={ isRtl ? wpToJpImageRtl : wpToJpImage }
-				width="49"
-				height="29"
+				width={ iconWidth }
+				height={ iconSize }
 				alt="WordPress and Jetpack app"
 			/>
 			<div className="app-promo__title">
-				<CardHeading tagName="h2">{ translate( 'Get our mobile app' ) }</CardHeading>
-				<h3 className="app-promo__subheader">
-					{ translate( 'Everything you need to build and grow your site from any device.' ) }
-				</h3>
+				<CardHeading tagName="h2">{ title }</CardHeading>
```

Comment:
Might be worth wrapping with [`preventWidows`](https://github.com/Automattic/wp-calypso/blob/acfaee33659330e5f9f4579bc18768ac12cdea2f/client/lib/formatting/prevent-widows.js#L9C17-L9C30) considering banner can be narrow on phones and different translations and strings can end up looking unbalanced.

Reply:
Good call!